src/core/store/useShoppingCartStore.ts

```diff
@@ -0,0 +1,83 @@
+import { create } from 'zustand';
+import { persist } from 'zustand/middleware';
+
+import type { Product } from '@/shared/types/productType';
+
+export type CartItem = Product & {
+    quantity: number;
+    color?: string;
+    size?: string;
+    // other custom props depending on the product
+};
+
+export interface CartState {
+    items: CartItem[];
+    addItem: (cartItem: CartItem) => void;
+    removeItem: (id: string) => void;
+    updateItem: (id: string, updatedItem: CartItem) => void;
+    updateQuantity: (id: string, quantity: number) => void;
+    clearCart: () => void;
+}
+
+export const useCartStore = create<CartState>()(
+    persist(
+        (set, get) => ({
+            items: [],
+
+            addItem: (cartItem) => {
+                const items = get().items;
+                const existingItem = items.find((i) => i.id === cartItem.id);
+
+                if (existingItem) {
+                    set({
+                        items: items.map((i) =>
+                            i.id === cartItem.id
+                                ? {
+                                      ...i,
+                                      quantity: i.quantity + cartItem.quantity,
+                                  }
+                                : i
+                        ),
+                    });
```

Comment:
Не люблю я ось ці супер короткі назви `i => item`
Приведу тут приклад, але краще скрізь це виправляти.

```suggestion
                    set({
                        items: items.map((item) =>
                            item.id === cartItem.id
                                ? {
                                      ...item,
                                      quantity: i.quantity + cartItem.quantity,
                                  }
                                : item
                        ),
                    });
```